src/main/java/org/jenkinsci/plugins/workflow/job/WorkflowJob.java

```diff
@@ -180,25 +180,15 @@ public void setDefinition(FlowDefinition definition) {
     @Override protected void submit(StaplerRequest req, StaplerResponse rsp) throws IOException, ServletException, Descriptor.FormException {
         super.submit(req, rsp);
         JSONObject json = req.getSubmittedForm();
-        // TODO: Switch from Descriptor.newInstancesFromHeteroList to Descriptor.bindJSON when minimum supported
-        //  version is Jenkins 2.342 or higher.
-        definition = json.get("definition") == null ? null :
-                Descriptor.newInstancesFromHeteroList(req, json.getJSONObject("definition"),
-                        ExtensionList.lookup(FlowDefinitionDescriptor.class)).get(0);
+        definition = req.bindJSON(FlowDefinition.class, json.getJSONObject("definition"));
```

Comment:
`StaplerRequest#bindJSON` isn't the same as `Descriptor#bindJSON`

Reply:
Yeah, this either needs to be reverted to match https://github.com/jenkinsci/workflow-job-plugin/commit/a25d15548158461749870d328746e2be482f9e4d or changed to use `Descriptor.bindJSON`. It looks like <a href="https://github.com/jenkinsci/workflow-job-plugin/pull/261/commits/a8af864562879f1dca792a51417a1439c7fb82c7">this merge commit</a> is where the problem was introduced.

Reply:
It is not the same but the idea was that https://github.com/jenkinsci/jenkins/pull/5417 could let us simplify code.

From local testing, using `req.bindJSON` works so long as you _also_ make the call to `Descriptor.newInstancesFromHeteroList` (even later). Something is using weird side effects.

Reply:
> Something is using weird side effects.

Yes, the call to `ScriptApproval.configuring` in `CpsFlowDefinition$DescriptorImpl.newInstance` is a side effect.

Either the code can be reverted to use `Descriptor.newInstancesFromHeteroList` or you can switch to `Descriptor.bindJSON` so that `CpsFlowDefinition$DescriptorImpl.newInstance` gets called.